src/components/task-item.tsx

```diff
@@ -86,9 +86,9 @@ export function TaskItem({ task }: { task: Task }) {
       upsertNote({
         id: task.noteId,
```

Comment:
Should we just return early if `note` is null?


```suggestion
      if (!note) return
      
      upsertNote({
        id: note.id,
```